### PR TITLE
Started adding a second room to highlight and fix issues

### PR DESCRIPTION
**Pull request by rachidlamouri**
Atomic commits ftw

- Found and fixed tests that only assume there is one room
- Found a bug in schema-to-generator where it wouldn't restrict the possible "enum" values
    - Example: setting type to "exampleRoom1" would still allow PlayerState values "State2A" and "State2B"
    -  I forgot that the new version of schema-to-generator cannot parse refs, so I had to fix that as well
        - let me know if you can find an existing **synchronous** JSON-schema dereferencing library (I can only find async ones)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,3 +49,15 @@ Tests use [semantic-mocha](./semantic-mocha/README.md) which is currently being
...
+
+1. Add a doc for the room in [docs/rooms/](./docs/rooms/)
+    - Make sure it's consistent with the other docs.
+1. Update [gameState.schema.json](./src/utils/types/gameState.schema.json):
+    1. Add a new schema to the "definitions" object for your room state
+    1. Add a $ref to "definitions.RoomState" that points to the new schema
+1. Compile the game state schemas with `npm run compile:gameStateSchema`
+1. Create a new subclass of [RoomHandler](./src/roomHandlers/roomHandler.ts)
+    1. Create a type alias to reference the room type within the sublcass (for easier maintenance)
+1. Update the "allRoomHandlersByRoomType" in [roomUtil](./src/utils//roomUtil.ts) to include the new room handler
```
Comment (jshemas) [inline README.md:63]: The link still works, but I think you meant to do `./src/utils/roomUtil.ts` here.

File: docs/rooms/exampleRoom3Room.md [hunk 2]
```diff
@@ -18,21 +18,21 @@ This room will go away when other rooms are implemented.
 
 ## Commands
 
-- **leave**: You leave example room 3
-- **go3A**: You move to state 3A
-- **go3B**: You move to state 3B
-- **goStart**: You move back to the entrance
+- **leave**: You leave example room 3 having completed n laps
```
Comment (jshemas) [inline docs/rooms/exampleRoom3Room.md:21]: I've been using `X` for vars. Example: https://github.com/normal-devs/klicker-knight/blob/main/docs/rooms/fishingRoom.md?plain=1#L7

Should we update this to `n`?

File: src/roomHandlers/exampleRoom2Handler.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      type: roomType,
+      playerState: 'AtEntrance',
+    };
+  }
+
+  protected getCommandHandler(
+    roomState: TRoomState,
+    command: Command,
+  ): TNullableCommandHandler {
+    if (command === 'leave') {
```
Comment (jshemas) [inline src/roomHandlers/exampleRoom2Handler.ts:46]: Looks like the rest of the commands are in https://github.com/normal-devs/klicker-knight/pull/30

File: docs/rooms/exampleRoom3Room.md (hunk 2 above)
Comment (rachidlamouri) [reply docs/rooms/exampleRoom3Room.md:21]: I don't see a need to make the variables consistent between docs. I just think we need to be consistent within docs. I assume at some point we will have rooms with multiple variables and IMO we should pick whatever references we think make sense for that room.

File: README.md (hunk 1 above)
Comment (rachidlamouri) [reply README.md]: Fixed

**Review by jshemas — APPROVED** [review]

**Pull request merged**